strings-csharp/HexToByteArray/HexToByteArray/ConvertHex.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace HexToByteArray
+{
+    public class ConvertHex
+    {
+        public byte[] FromHexString(string hex)
+        {
+            try
+            {
+                return Convert.FromHexString(hex);
+            }
+            catch (Exception)
+            {
+
+                throw;
+            }
+        }
+
+        public byte[] FromHexStringAlternative(string hex)
+        {
+            try
+            {
+                int length = hex.Length;
+                byte[] bytes = new byte[length / 2];
+                for (int i = 0; i < length; i += 2)
+                    bytes[i / 2] = Convert.ToByte(hex.Substring(i, 2), 16);
```

Comment:
You should be using a span here instead of Substring. This call is creating lots of small temporary strings. If you added a benchmark that used a longer string, something like 1024 characters or 2048 you would really start to see the effect of this. By taking the string as a ReadOnlySpan you can just slice the characters you want (span.Slice(i, 2)). Instead of Convert.ToByte you can use byte.Parse or TryParse. With Parse it will just throw an exception if the characters aren't in a proper format. Note that you have to provide the NumberStyles.HexNumber for Parse to work properly.